sphinx/builders/linkcheck.py

```diff
@@ -154,9 +152,126 @@ def redirected(self) -> Dict[str, Tuple[str, int]]:
             RemovedInSphinx50Warning,
             stacklevel=2,
         )
-        return self._redirected
+        return {}
 
     def check_thread(self) -> None:
+        warnings.warn(
+            "%s.%s is deprecated." % (self.__class__.__name__, "redirected"),
+            RemovedInSphinx50Warning,
+            stacklevel=2,
+        )
+        # do nothing.
+
+    def limit_rate(self, response: Response) -> Optional[float]:
+        warnings.warn(
+            "%s.%s is deprecated." % (self.__class__.__name__, "redirected"),
+            RemovedInSphinx50Warning,
+            stacklevel=2,
+        )
+        return HyperlinkAvailabilityCheckWorker(self).limit_rate(response)
+
+    def process_result(self, result: Tuple[str, str, int, str, str, int]) -> None:
```

Comment:
Note: The methods below are not changed at all. I only moved (copied) `check_thread()` and `limit_rate()` to `HyperlinkAvailabilityCheckWorker` class.